suite/suite_test.go

```diff
@@ -135,10 +151,22 @@ func TestMain(t *testing.T) {
 
 	for r := 1; r <= *repeat; r++ {
 		for _, osFlavor := range osFlavors {
-			for ts, entry := range testSet {
-				for _, drv := range storageDrivers {
+			var osVendor, osVersion = osFlavor, ""
+			if split := strings.Split(osFlavor, ":"); len(split) == 2 {
+				osVendor = split[0]
+				osVersion = split[1]
+			}
+			for _, drv := range storageDrivers {
+				if !flavorSupported(osVendor, osVersion, drv) {
+					continue
+				}
+				for ts, entry := range testSet {
 					cfg := config.WithTag(fmt.Sprintf("%s-%d", ts, r)).
-						WithOS(osFlavor).WithStorageDriver(drv)
+						WithOS(osVendor)
```

Comment:
this could have been `WithOS(osFlavor).WithStorageDriver(driver)`

Reply:
then it'll make a weird tag name of `centos-overlay2-7.3` instead of `centos-7.3-overlay2`